.github/workflows/release.yaml

```diff
@@ -0,0 +1,45 @@
+# Release workflow: Pushing a tag triggers this workflow,
+# which uses the name of the tag as the container tag/version
+# that goes into GHCR.
+name: Release
+
+# Build and publish Docker image upon pushing a tag
+on:
+  push:
+    tags:
+      - "v*"
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        # actions/checkout@v2 breaks annotated tags by converting them into
+        # lightweight tags, so we need to force fetch the tag again
+        # See: https://github.com/actions/checkout/issues/290
+      - name: "Repair tag"
+        run: git fetch -f origin ${{ github.ref }}:${{ github.ref }}
+      - name: "Verify tag is annotated"
+        run: if test x$(git for-each-ref ${{ github.ref }} | awk '{print $2}') = xtag; then /bin/true; else echo "\"${{ github.ref }}\" does not look like an annotated tag!"; /bin/false; fi
+      - name: "Echo release tag"
+        run: echo "TAG=${{ github.repository }}:${{ github.ref }}"
+      - name: "Log in to Docker"
+        uses: docker/login-action@v1
+        with:
+          password: "${{ secrets.GITHUB_TOKEN }}"
+          registry: ghcr.io
+          username: "${{ github.actor }}"
+      - id: docker_build
+        name: "Build the release Docker image"
+        uses: docker/build-push-action@v2
+        with:
+          push: false
+          tags: "${{ github.repository }}:${{ github.ref }}"
+      - name: "Tag the build for GHCR"
+        run: docker tag docker.io/${{ github.repository }}:${{ github.ref }} ghcr.io/${{ github.repository }}:${{ github.ref }}
```

Comment:
What is going on with this step and the following one? Why don't we just let the docker/build-push-action push and tag the image?

If nothing else, the "docker.io" here seems odd. We aren't pushing anything to docker.io, are we?

Reply:
Great point not sure why it was done that way originally either, I updated it to push all tags using that action.